data-raw/wales-ltla-summary-metrics.R

```diff
@@ -0,0 +1,121 @@
+library(tidyverse)
+library(IMD)
+library(geographr)
+library(sf)
+library(ggridges)
+library(stringr)
+
+ltla <- boundaries_ltla21 %>%
+  st_drop_geometry() %>%
+  filter(str_detect(ltla21_code, "^W")) %>%
+  select(ltla21_code,ltla21_name)
+
+lookup_ms <- lookup_msoa11_ltla21 %>%
+  select(msoa11_code, ltla21_code) %>%
+  filter(str_starts(ltla21_code, "W") | str_starts(msoa11_code, "W"))
+
+# ---- IMD score ----
+# Higher extent = more deprived
+# Higher rank  = more deprived
+imd <- imd_wales_lad %>%
+  select(ltla21_code = lad_code, imd_score = Extent) %>%
+  mutate(number = rank(imd_score)) %>%
+  select(-imd_score) %>%
+  mutate(variable = "Index of Multiple \nDeprivation rank", .after = ltla21_code) %>%
+  mutate(percent = NA, .after = number)
+
+#### ---- % Left-behind areas ----
+# Higher number/percent = more left-behind
+lba <- cni_wales_msoa11 %>%
+  right_join(lookup_ms, by = "msoa11_code") %>%
+  select(ltla21_code, lba = `Left Behind Area?`) %>%
+  group_by(ltla21_code) %>%
+  count(lba) %>%
+  mutate(percent = n / sum(n)) %>%
+  filter(lba == TRUE) %>%
+  right_join(ltla, by = "ltla21_code") %>%
+  mutate(percent = replace_na(percent, 0)) %>%
+  mutate(number = replace_na(n, 0)) %>%
+  select(ltla21_code, number, percent) %>%
+  mutate(variable = "Left-behind areas", .after = ltla21_code)
+
+# ---- Health Index Score ----
+#  Use the BRC Resilience on git hub
+# Higher score = worse health
+# Higher rank  = worse health
+url <- "https://raw.githubusercontent.com/britishredcrosssociety/resilience-index/main/data/vulnerability/health-inequalities/wales/healthy-people-domain.csv"
+health_index_raw <- read_csv(url)
+health_index <- health_index_raw |>
+  cross_join(lookup_ms) |>
```

Comment:
As far as I can tell, the `cross_join()` has no means to match rows by a key, and so is a dangerous option as assumes the rows are in the correct order. It also results in two local authority code columns. Please replace with a `left_join()`, specifying a joining key:

```r
left_join(lookup_ms, by = c("lad_code" = "ltla21_code"))
```